Modules/Forum/classes/class.ilForumPostDraft.php

```diff
@@ -384,7 +387,52 @@ protected static function readDrafts($user_id)
             self::$instances[$user_id]['draft_ids'][$tmp_obj->getDraftId()] = $tmp_obj;
         }
     }
+
+    /**
+     * @param int $user_id
+     * @param int $thread_id
+     * @param int $sorting
+     * @return ilForumPostDraft[]
+     */
+    public static function readSortedDrafts(int $user_id, int $thread_id, int $sorting = ilForumProperties::VIEW_DATE_ASC) : array
+    {
+        global $DIC;
+        $ilDB = $DIC->database();
+    
+        $drafts = [];
+    
+        $order_statement = ' ';
+        $order_direction = ' ';
     
+        if ($sorting == ilForumProperties::VIEW_DATE_ASC || $sorting == ilForumProperties::VIEW_DATE_DESC) {
+            $order_statement = ' ORDER BY post_date ';
+            $order_direction = 'ASC';
+            if ($sorting == ilForumProperties::VIEW_DATE_DESC) {
+                $order_direction = 'DESC';
+            }
+        }
+        $res = $ilDB->queryF(
+            'SELECT * FROM frm_posts_drafts WHERE post_author_id = %s AND thread_id = %s' .
+            $order_statement . $order_direction,
+            ['integer', 'integer'],
+            [$user_id, $thread_id]
+        );
+    
+        while ($row = $ilDB->fetchAssoc($res)) {
+            $draft = new ilForumPostDraft();
+            self::populateWithDatabaseRecord($draft, $row);
+            $drafts[] = $draft;
+            self::$instances[$user_id][$thread_id][$draft->getPostId()][] = $draft;
+        }
+    
+        if ($sorting == ilForumProperties::VIEW_TREE) {
```

Comment:
@nmatuschek Why do we have this different return statements here?

Reply:
The rendering process is handled differently in the GUI dependent to the viewmode. While the rendering by date does not need a parent node, the hierarchical rendering by tree does. The parent post node can have multiple drafts beneath, that's why the return of an array with multiple drafts is needed here.   
For the rendering by date we know, the it is also possible, that in some cases there is no parent post id needed i.e. the dummy node. For now, I did not found a practicable solution to unify this process without the need to change the whole rendering process itself. :-( not happy with it. In fact we need a thoughtful concept for this.

Reply:
Okay, I see. It's okay for me for this bugfix, but as you said: We'll have to consider this (refactoring) for (possible) future features.